src/core/demangle.d

```diff
@@ -296,6 +325,13 @@ private struct Demangle
         return char.init;
     }
 
+    char peek( size_t n )
+    {
+        if( pos + n < buf.length )
+            return buf[pos + n];
```

Comment:
I speculate speed is important here - perhaps you can make this trusted and skip bounds checking.